### PR TITLE
[MOB-15818] Added documentation directory with getting started guide and api reference pages

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+```java
+String extensionVersion = Identity.extensionVersion();
+```
+------
+
+### getExperienceCloudId
+
+This API retrieves the Experience Cloud ID (ECID) that was generated when the app was initially launched. This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall.
+
+> The ECID value is returned via the [AdobeCallback](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#public-classes). When [AdobeCallbackWithError](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#public-classes) is provided to this API, the timeout value is 500ms. If the operation times out or an unexpected error occurs, the `fail` method is called with the appropriate [AdobeError](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#public-classes).
```
Comment (emdobrin) [inline Documentation/api-reference.md:43]: I am thinking we should remove the external links to the MobileCore classes, and use simple code like quotes instead, e.g. `AdobeCallbackWithError`

File: Documentation/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
+# Adobe Experience Platform Identity for Edge Network Extension - Android
+
+## Prerequisites
+
+Refer our [Getting Started Guide](getting-started.md)
```
Comment (emdobrin) [inline Documentation/api-reference.md:5]: Refer to the [getting started guide](getting-started.md).

File: Documentation/api-reference.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    public void call(IdentityMap identityMap) {        
+         //Handle the IdentityMap returned here    
+    }
+});
+```
+
+------
+
+### getUrlVariables
+> :information_source: This method was added in Edge Identity version 1.1.0.
```
Comment (emdobrin) [inline Documentation/api-reference.md:93]: This API is available with version 1.1.0 and above.

File: Documentation/api-reference.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+```java
+Identity.getUrlVariables(new AdobeCallback<String>() {    
+    @Override    
+    public void call(String urlVariablesString) {        
+        //handle the URL query parameter string here
+        //For example, open the URL on the device browser        
+        //        
+        Intent i = new Intent(Intent.ACTION_VIEW);        
+        i.setData(Uri.parse("https://example.com?" + urlVariablesString));        
+        startActivity(i);
```
Comment (emdobrin) [inline Documentation/api-reference.md:122]: Can we update this example to use webView.loadUrl https://developer.android.com/reference/android/webkit/WebView#loadUrl(java.lang.String) ?

File: Documentation/api-reference.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+------
+
+### registerExtension
+
+Registers the Identity for Edge Network extension with the Mobile Core extension.
+
+> :information_source: If your use-case covers both Edge Network and Adobe Experience Cloud Solutions extensions, you need to register Identity for Edge Network and Identity for Experience Cloud Identity Service from Mobile Core extensions. For more details, see the [frequently asked questions](https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network/identity-faq#q-i-am-using-aep-edge-and-adobe-solutions-extensions-which-identity-extension-should-i-install-and-register).
+
+The extension registration occurs by passing Identity for Edge Network extension to the [MobileCore.registerExtensions API](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#registerextension-s).
```
Comment (emdobrin) [inline Documentation/api-reference.md:135]: this is not accurate for the Android implementation, let's remove

**Review by emdobrin — APPROVED** [review]

**Review by cacheung — APPROVED** [review]

**Pull request merged**